python/llm/src/ipex_llm/transformers/low_bit_linear.py

```diff
@@ -229,8 +230,12 @@ def ggml_convert_qtype(tensor: torch.Tensor, qtype: int,
         scale = torch.empty(n // k, dtype=torch.float32,
                             device=device)
     else:
-        dst_tensor = torch.empty(dst_size, dtype=torch.uint8,
-                                 device=device)
+        if enable_deepspeed_zero3:
+            dst_tensor = torch.empty(dst_size // 2, dtype=torch.bfloat16,
+                                     device=device)
```

Comment:
I think we should always do that for NF4 (only)?

Reply:
Other NF4s are packed in torch.uint8, which do not make the buffer length redundant.
Only deepspeed zero3 needs NF4 to be packed in torch.bfloat16, which needs to halve the buffer.